pkg/http/listener.go

```diff
@@ -107,6 +96,30 @@ func (listener *httpListener) start() {
 		}
 	}
 
+	// Guess TLS protocol returns true if HTTP1.1 or
+	// HTTP2 methods are not found in the input buffer.
+	//
+	// This function if it returns true doesn't mean
+	// buffer has indeed TLS data, which needs
+	// to be further validated by using tls handshake.
+	guessTLSProtocol := func(buf []byte) bool {
```

Comment:
You are doing http method check whereas you named `guessTLSProtocol`?  What is the problem with `isHTTPMethod()`?